recipes-multimedia/kodi-systemd-service/kodi-systemd-service.bb

```diff
@@ -5,20 +5,16 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
 
 inherit kodi-common systemd
 
-SRC_URI = " \
-  file://kodi-x11.service \
-  file://kodi.service \
-"
+PREFFERED_KODI_SERVICE ?= "kodi-${@bb.utils.contains('KODI_GRAPHICAL_BACKEND', 'x11', 'x11', 'gbm', d)}.service"
 
-SYSTEMD_SERVICE_${PN} = "kodi.service"
-SYSTEMD_AUTO_ENABLE_${PN} = "${KODI_SYSTEMD_AUTOSTART}"
+SRC_URI = "file://${PREFFERED_KODI_SERVICE}"
 
-do_install_append() {
-	install -d ${D}${systemd_unitdir}/system
+SYSTEMD_SERVICE_${PN} = "${PREFFERED_KODI_SERVICE}"
+SYSTEMD_AUTO_ENABLE_${PN} = "${PREFFERED_KODI_SERVICE}"
+
+FILES_${PN} = "${systemd_unitdir}/system"
 
-	if [ -e ${D}${libdir}/kodi/kodi-gbm ] ; then
```

Comment:
This doesn't work properly since this is separate recipe. I reworked recipe, but I think this is a temporary approach, it may change in near feature. I'll think about this.